### PR TITLE
feat(hybridcloud) First pass at a siloed devserver

### PR DESCRIPTION
**Pull request by markstory**
With foreign keys mostly broken, we can start building the siloed development environment. While this doesn't include all the desired functionality it has working webservers + and a functioning RPC bridge.

### What's included

* Webserver for control + region silos
* A simple way to start `SENTRY_USE_SILOS=1 sentry devserver`
* Webpack proxying for both control and region silos.
* These changes also contain the changes from #49922

### Getting started

Before one can use siloed development server, you'll first need to generate the split databases from your current monolith database using `make create-db` and `bin/split-silo-database`. Then you'll be able to start the servers.

### What's not working

* Sentry apps & integrations still contain many silo containment errors.
* Organization member operations still contain failures as we've not finished fully separating the Organizationmember : User relationship.

### What will be fixed next

* Partitioning celery beat jobs during startup.
* Fixing up the various silo assignment issues.
* Expanding the control silo proxy paths.

### CODE REVIEW

File: src/sentry/auth/superuser.py [hunk 1]
```diff
@@ -131,6 +131,9 @@ def is_active(self):
         org = getattr(self.request, "organization", None)
         if org and org.id != self.org_id:
             return self._check_expired_on_org_change()
+        # We have a wsgi request with no user.
+        if not hasattr(self.request, "user"):
```
Comment (mikejihbe) [inline src/sentry/auth/superuser.py:135]: Is this the case for RPC calls?

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -3297,6 +3299,40 @@ def build_cdc_postgres_init_db_volume(settings):
...
+                "name": "us",
+                "snowflake_id": 1,
+                "category": "MULTI_TENANT",
+                "address": "http://localhost:8000",
+                "api_token": "dev-region-silo-token",
+            }
+        ]
+    )
+    control_port = os.environ.get("SENTRY_CONTROL_SILO_PORT", "8010")
+    DEV_HYBRID_CLOUD_RPC_SENDER = json.dumps(
```
Comment (mikejihbe) [inline src/sentry/conf/server.py:3327]: This...seems like a bad name for this. I had to look up usage to understand what it was for. Unrelated to your changeset, I realize, but cc @RyanSkonnord

File: src/sentry/runner/commands/devserver.py [hunk 3]
```diff
@@ -201,6 +201,8 @@ def devserver(
     needs_https = parsed_url.scheme == "https" and (parsed_url.port or 443) > 1024
     has_https = shutil.which("https") is not None
 
+    control_silo_port = port + 10
```
Comment (mikejihbe) [inline src/sentry/runner/commands/devserver.py:204]: should this use the same env var from above if it's set? `SENTRY_CONTROL_SILO_PORT`

File: webpack.config.ts [hunk 4]
```diff
@@ -555,6 +556,31 @@ if (
...
+    // If we're running siloed servers we also need to proxy
+    // those requests to the right server.
+    let controlSiloProxy = {};
+    if (CONTROL_SILO_PORT) {
+      // TODO(hybridcloud) We also need to use this URL pattern
+      // list to select contro/region when making API requests in non-proxied
+      // environments (like production). We'll likely need a way to consolidate this
+      // with the configuration api.Client uses.
+      const controlSiloAddress = `http://127.0.0.1:${CONTROL_SILO_PORT}`;
+      controlSiloProxy = {
```
Comment (mikejihbe) [inline webpack.config.ts:568]: 🔥

**Review by mikejihbe — APPROVED** [review]

File: src/sentry/auth/superuser.py (hunk 1 above)
Comment (markstory) [reply src/sentry/auth/superuser.py:135]: Yes, I had to remove authentication (and user loading) for them as the token support we have today requires using an access token from a user. We shouldn't be using a user bound token for these operations though.

**Pull request merged**